File: Data Structures & Algorithms/special-array-with-x-elements-greater-than-or-equal-x/submission-0.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        int n = nums.size();
        std::vector<int> counts(n + 1, 0);
        for(int num:nums){
            if(num>=n)
                counts[n]++;
            else
                counts[num]++;
        }
        int x=0;
        for(int j=n;j>=0;j--){
            x+=counts[j];
            if(x==j)
                return j;
        }
    return -1;
    }
};

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        subs=[[]]
        for num in nums:
            curr_len = len(subs)
            for j in range(curr_len):
                subs.append(subs[j] + [num])
        return subs